#!/usr/bin/env python
PACKAGE = "duo3d_ros"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("gain",       double_t,  0, "The gain level on the DUO3D",     10, 0, 100)
gen.add("exposure",   double_t,  0, "The exposure level on the DUO3D", 60, 0, 100)
gen.add("LED",	      double_t,  0, "The LED level on the DUO3D",      20, 0, 100)
gen.add("CameraSwap",   bool_t,  0, "Sets camera swap value", 		False)
gen.add("HorizontalFlip",bool_t, 0, "Horizontally flip the image",      False)
gen.add("VerticalFlip",  bool_t, 0, "Vertically flip the image",        True)
gen.add("speckleWindowSize",       int_t,  0, "Speckle Window Size (dense3D)",     60, 0, 256)
gen.add("speckleRange",   int_t,  0, "Speckle Range (dense3D)", 8, 0, 16)
gen.add("uniquenessRatio",	      int_t,  0, "Uniqueness Ratio (dense3D)",      26, 0, 100)
gen.add("preFilterCap",   int_t,  0, "Pre-filter cap (dense3D)", 		20,0, 256)
gen.add("sadWindowSize",   int_t,  0, "Pre-filter cap (dense3D)", 		5,2, 10)
gen.add("numDisparities",   int_t,  0, "Number of disparities (dense3D)", 		3,2, 16)
gen.add("dense3dMode",   int_t,  0, "Mode (0 - BM, 1 - SGBM)", 		1,0,1)

exit(gen.generate(PACKAGE, "duo3d_ros", "Duo"))
